src/traces/barpolar/plot.js

```diff
@@ -0,0 +1,127 @@
+*/
+
+'use strict';
+
+var d3 = require('d3');
+var isNumeric = require('fast-isnumeric');
+
+var Lib = require('../../lib');
+var Drawing = require('../../components/drawing');
+var helpers = require('../../plots/polar/helpers');
+
+module.exports = function plot(gd, subplot, cdbar) {
+    var xa = subplot.xaxis;
+    var ya = subplot.yaxis;
+    var radialAxis = subplot.radialAxis;
+    var angularAxis = subplot.angularAxis;
+    var pathFn = makePathFn(subplot);
+    var barLayer = subplot.layers.frontplot.select('g.barlayer');
+
+    Lib.makeTraceGroups(barLayer, cdbar, 'trace bars').each(function(cd) {
+        var cd0 = cd[0];
+        var plotGroup = cd0.node3 = d3.select(this);
+        var t = cd0.t;
+
+        var poffset = t.poffset;
+        var poffsetIsArray = Array.isArray(poffset);
+
+        var pointGroup = Lib.ensureSingle(plotGroup, 'g', 'points');
+        var bars = pointGroup.selectAll('g.point').data(Lib.identity);
+
+        bars.enter().append('g')
+            .classed('point', true);
+
+        bars.exit().remove();
+
+        bars.each(function(di, i) {
+            var bar = d3.select(this);
+
+            // TODO move this block to Bar.setPositions?
+            //
+            // now display the bar
+            // clipped xf/yf (2nd arg true): non-positive
+            // log values go off-screen by plotwidth
+            // so you see them continue if you drag the plot
+            //
+            // this gets reused in ./hover.js
+            var p0 = di.p0 = di.p + ((poffsetIsArray) ? poffset[i] : poffset);
+            var p1 = di.p1 = p0 + di.w;
+            var s0 = di.s0 = di.b;
+            var s1 = di.s1 = s0 + di.s;
+
+            var rp0 = radialAxis.c2p(s0);
+            var rp1 = radialAxis.c2p(s1);
+            var thetag0 = angularAxis.c2g(p0);
+            var thetag1 = angularAxis.c2g(p1);
+
+            var dPath;
+
+            if(!isNumeric(rp0) || !isNumeric(rp1) ||
+                !isNumeric(thetag0) || !isNumeric(thetag1) ||
+                rp0 === rp1 || thetag0 === thetag1
+            ) {
+                // do not remove blank bars, to keep data-to-node
+                // mapping intact during radial drag, that we
+                // can skip calling _module.style during interactions
+                dPath = 'M0,0Z';
+            } else {
+                // TODO is this where we want for to select for barpolar?
+                var rg1 = radialAxis.c2g(s1);
+                var thetagMid = (thetag0 + thetag1) / 2;
+                di.ct = [
+                    xa.c2p(rg1 * Math.cos(thetagMid)),
+                    ya.c2p(rg1 * Math.sin(thetagMid))
+                ];
+
+                // TODO round up bar borders?
+                // if so, factor out that logic from Bar.plot
+
+                dPath = pathFn(rp0, rp1, thetag0, thetag1);
+            }
+
+            Lib.ensureSingle(bar, 'path')
+                .style('vector-effect', 'non-scaling-stroke')
```

Comment:
do we need a `stroke-miterlimit` here? `pie` uses [`stroke-linejoin: 'round'`](https://github.com/plotly/plotly.js/blob/19f3800bac848e702948fcc32302ded56ea1cdb7/src/traces/pie/plot.js#L34-L36) as you can see if you give pies a very wide linewidth... but I think we probably want more precision here, so a regular mitered corner with a large enough `stroke-miterlimit` that the 90º corners are not clipped, but small enough that two neighboring bars that meet at the center with wide line width, no matter what angle the bars subtend, will never show crossing tips.

Reply:
Playing around with `stroke-miterlimit`, I couldn't get the mocks to produce a diff. So it's a little hard to tell if it improves the bar's appearance. Perhaps this is due to us using (old) nw.js. It might be worth trying different  `stroke-miterlimit` value in orca.

Reply:
Here is the effect I was worried about:  two slices meet at a sharp point, but the strokes end up making them look like they cross. For the angles in the codepen below it only seems to be a problem if I insert a large `stroke-miterlimit`, ~7 or above:
https://codepen.io/alexcjohnson/pen/WgOpqP
![screen shot 2018-09-04 at 12 19 44 pm](https://user-images.githubusercontent.com/2678795/45044035-efa69c00-b03c-11e8-9413-fe52c96f2bb9.png)
but for other angles (somewhat wider but not too wide) the [default value of 4](https://developer.mozilla.org/en-US/docs/Web/SVG/Attribute/stroke-miterlimit) might be problematic. For scatter traces I [set it to 2](https://github.com/plotly/plotly.js/blob/854bbc931c218be579850b45a04b666d40fc53eb/src/traces/scatter/plot.js#L39) mostly so that very spiky traces don't look like they peak higher than they really do. I suspect 2 would be a good value here too, it's conceivable the default of 4 is fine or has such uncommon failures that we wouldn't care.

Reply:
`2` seems like to way to go -> https://github.com/plotly/plotly.js/pull/2954/commits/183b73490bf5eabc1914efefb451d731e48771b8